src/Sylius/Bundle/CoreBundle/Doctrine/ORM/OrderRepository.php

```diff
@@ -419,4 +420,16 @@ public function findCartForSelectingPayment($id): ?OrderInterface
 
         return $order;
     }
+
+    public function findOneByTokenValue(string $tokenValue): ?BaseOrderInterface
```

Comment:
This is confusing IMHO, cause inside you force to return only in `cart` state, maybe should be renamed to: `findCartByTokenValue()`?